### PR TITLE
ringbuf: Allow poll() to block for an indefinite time period

### PR DESCRIPTION
**Pull request by insearchoflosttime**
ringbuf: Allow poll() to block for an indefinite time period

    ring_buffer__poll in libbpf internally calls epoll_wait. Specifying a
    timeout of -1 to epoll_wait causes it to block indefinitely if no event
    transpires. We should support this functionality in libbpf-rs as well.

    In this change, if the user passes in Duration::MAX as the timeout, this
    will cause the ringbuf poll to block indefinitely until an event occurs.

### CODE REVIEW

**Review by anakryiko — APPROVED** [review]
kind of unfortunate that all existing users will have to update their use of ringbuf, but oh well

**Comment by insearchoflosttime** [comment]
An alternative would be to introduce a new function like epoll_indefinite

**Comment by danobi** [comment]
Couldn't you check if the value is `Duration::MAX` and pass -1 if it is? Not sure we need an API break for this

**Comment by anakryiko** [comment]
> Couldn't you check if the value is `Duration::MAX` and pass -1 if it is? Not sure we need an API break for this

@insearchoflosttime did you get a chance to try @danobi's suggestion with Duration::MAX ?

**Comment by insearchoflosttime** [comment]
@anakryiko  yes! I will rebase and push it out today.

File: libbpf-rs/src/ringbuf.rs [hunk 1]
```diff
@@ -136,7 +136,12 @@ impl<'a> RingBuffer<'a> {
     pub fn poll(&self, timeout: Duration) -> Result<()> {
         assert!(!self.ptr.is_null());
 
-        let ret = unsafe { libbpf_sys::ring_buffer__poll(self.ptr, timeout.as_millis() as i32) };
+        let mut timeout_ms = -1;
+        if timeout != Duration::MAX {
+            timeout_ms = timeout.as_millis() as i32;
```
Comment (anakryiko) [inline libbpf-rs/src/ringbuf.rs:141]: looks good, but let's update the doc comment for poll() to specify that Duration::MAX is used to mean "infinite timeout"?

**Pull request merged**